### ISSUE TITLE
Global seed(random & numpy) may change?

### ISSUE THREAD

**Issue by noprohuangqi** [open-mmlab/mmdetection#4193]
Firstly , Thanks for your work.
Question is that if we set a global random seed as 42, then I find when dataloader work, it will change the seed.
```
def worker_init_fn(worker_id, num_workers, rank, seed):
    # The seed of each worker equals to
    # num_worker * rank + worker_id + user_seed
    worker_seed = num_workers * rank + worker_id + seed
    np.random.seed(worker_seed)
    random.seed(worker_seed)
```
In this situation, some feature may not work as normal.
For example, we set a global random seed as 42 , then 
1.a model train for 10epoch 
2.a model train for 1epoch & resume 9epoch ,
the result may be different, because when we resume from 1epoch, the global seed is 42; but if without resume , the global seed after 1epoch is not 42.

**Comment by yhcao6** [comment]
Setting the seed is mainly aimed at reducing the fluctuation but it can't guarantee you will get the same results after each experiment even without interruption. For detail, you can refer https://pytorch.org/docs/stable/notes/randomness.html
